odoo/tools/misc.py

```diff
@@ -1564,6 +1564,16 @@ def __getattr__(self, attrib):
         return DotDict(val) if type(val) is dict else val
 
 
+class Namespace(DotDict):
+    """ Namespace that supports both .dot and ["dot"] notations """
```

Comment:
why this need?

Reply:
To be retrocompatible with the previous werkzeug session which supported both .dot and ["dot"] :)

Reply:
Searching 11100 files for "request\.session\[" (regex, case sensitive)
80 matches across 35 files

Some interesting matches, they would require we use `getattr` if we only supported the dot-notation:
* `request.session['identity-check-last'] = time.time()` (dash)
* `request.session['view_quote_%s' % order_sudo.id] = now` (string interpolation)
* `request.session[history] = orders.ids[:100]` (variable)